src/microbe_stage/MicrobeAI.cs

```diff
@@ -9,385 +10,179 @@
+                microbe.EngulfSize && DistanceFromMe(possiblePrey.Translation) < 10.0f * microbe.EngulfSize;
+            prey = possiblePrey.Translation;
         }
 
-        switch (lifeState)
+        if (microbe.IsBeingEngulfed)
         {
-            case LifeState.PLANTLIKE_STATE:
-                // This ai would ideally just sit there, until it sees a nice opportunity pop-up unlike neutral,
-                // which wanders randomly (has a gather chance) until something interesting pops up
-                break;
-            case LifeState.NEUTRAL_STATE:
-            {
-                // Before these would run every time, now they just run for the states that need them.
-                boredom = 0;
-                preyPegged = false;
-                prey = null;
-                if (predator == null)
-                {
-                    GetNearestPredatorItem(data.AllMicrobes);
-                }
-
-                // Peg your prey
-                if (!preyPegged)
-                {
-                    prey = null;
-                    prey = GetNearestPreyItem(data.AllMicrobes);
-                    if (prey != null)
-                    {
-                        preyPegged = true;
-                    }
-                }
-
-                targetChunk ??= GetNearestChunkItem(data.AllChunks);
-
-                EvaluateEnvironment(random);
-                break;
-            }
-
-            case LifeState.GATHERING_STATE:
-            {
-                // In this state you gather compounds
-                if (RollCheck(SpeciesOpportunism, 400.0f, random))
-                {
-                    lifeState = LifeState.SCAVENGING_STATE;
-                    boredom = 0;
-                }
-                else
-                {
-                    DoRunAndTumble(random);
-                }
-
-                break;
-            }
-
-            case LifeState.FLEEING_STATE:
-            {
-                if (predator == null)
-                {
-                    GetNearestPredatorItem(data.AllMicrobes);
-                }
-
-                // In this state you run from predatory microbes
-                if (predator != null)
-                {
-                    DealWithPredators(random);
-                }
-                else
-                {
-                    if (RollCheck(SpeciesActivity, 400, random))
-                    {
-                        lifeState = LifeState.PLANTLIKE_STATE;
-                        boredom = 0;
-                    }
-                    else
-                    {
-                        lifeState = LifeState.NEUTRAL_STATE;
-                    }
-                }
-
-                break;
-            }
-
-            case LifeState.PREDATING_STATE:
+            SetMoveSpeed(Constants.AI_BASE_MOVEMENT);
+        }
+        else if (predator != null &&
+            DistanceFromMe((Vector3)predator) < (1500.0 * SpeciesFear / Constants.MAX_SPECIES_FEAR))
```

Comment:
Use `predator.Value` to access the value of nullable variables.

Reply:
This problem is also in other places, but I'll just comment on this once.

Reply:
Do you have an example of where this codebase does this? This sounds to me like you want me to use try/catch as a null check.

Reply:
I didn't check, but I'm pretty sure nullable has the following properties: `HasValue` and `Value` which you can use to 1) check if it has a value 2) access the value directly without requiring a cast (this throws an exception if there isn't a value and you didn't check HasValue first).